test/transcoding/bookstore-server.cc

```diff
@@ -99,22 +114,28 @@ class BookstoreServiceImpl : public Bookstore::Service {
 
   // Bookstore::Service implementation
 
-  ::grpc::Status ListShelves(::grpc::ServerContext*,
+  ::grpc::Status ListShelves(::grpc::ServerContext* ctx,
                              const ::google::protobuf::Empty* request,
                              ListShelvesResponse* reply) {
     std::cerr << "GRPC-BACKEND: ListShelves" << std::endl;
     PrintRequest(*request);
+    if (printmetadata_) {
```

Comment:
Just a idea.   you can pass ctx to PrintRequest, change PrintRequest to also print metadata which is controlled by the global flag.  This is much cleaner.

Reply:
Done.